fluxc/src/main/java/org/wordpress/android/fluxc/store/ThemeStore.java

```
@@ -262,6 +303,22 @@ private void handleCurrentThemeFetched(@NonNull FetchedCurrentThemePayload paylo
         emitChange(event);
     }
 
+    private void searchThemes(@NonNull String searchTerm) {
+        mThemeRestClient.searchThemes(searchTerm);
+    }
+
+    private void handleThemesSearched(@NonNull SearchedThemesPayload payload) {
+        OnThemesSearched event = new OnThemesSearched(payload.themes);
```

If there's some error in this RestClient endpoint execution, for some reason when instantiating `OnThemesSearched` the error is not propagated, thus the line right below `event.isError()` returns `false` when it should really be returning `true`. Furthermore, a crash happens if execution is continued after that (because there's a null `payload.themes` access in the `for` loop right below).

In order to reproduce:
1. start the app
2. tap on `Themes` button
3. enter something i.e. `edin` in the search box
4. turn airplane mode ON
5. tap on `Search Themes` button
6. observe the crash happening.

---

Ok so [this other comment](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/573/files#r143072324) I made is actually what happens when there's an error, but the true cause of the crash is what I explained a little bit more up here

---

🤦‍♂️ this is a silly mistake. Should be `payload.isError()`!